src/main/java/software/amazon/awssdk/crt/s3/S3ClientOptions.java

```diff
@@ -346,4 +347,25 @@ public S3ClientOptions withS3ExpressCredentialsProviderFactory(S3ExpressCredenti
     public S3ExpressCredentialsProviderFactory getS3ExpressCredentialsProviderFactory() {
         return s3expressCredentialsProviderFactory;
     }
+
+    /**
+     * The amount of memory the CRT client is allowed to use.
+     * The client makes a best-effort attempt at memory limiting but might exceed this limit in some cases.
+     * If not provided, the client calculates this optimally from other settings, such as targetThroughput.
+     * @param memoryLimitBytes Memory limit in bytes. It should be at least 1GiB.
```

Comment:
can you mention that on 64 bit the default will be between 2GB and 8GB based on target throughput